### PR TITLE
portal(feature): view app list

### PR DESCRIPTION
**Pull request by illichso**
## Issue

https://gravitee.atlassian.net/browse/APIM-5463

## Description

As an API publisher

I want to view the list of applications of which I am a member

so that I can easily see what applications I belong to

## Additional context

https://github.com/user-attachments/assets/2eedc9b3-b00f-4ddf-a194-93883074af33

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-eypujojmsw.chromatic.com)

### CODE REVIEW

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     See the License for the specific language governing permissions and
     limitations under the License.
 
 -->
-<p>applications works!</p>
+
+@if (applicationPaginator$ | async; as applications) {
+  @if (applications.data.length === 0) {
+    <mat-card id="no-apps" appearance="outlined"> </mat-card>
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: Should have message here, no? Or will that be in a separate PR?

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss [hunk 2]
```diff
@@ -1,15 +1,56 @@
...
+  gap: 32px;
+}
+
+.app-list__container {
+  display: grid;
+  gap: 32px;
+  grid-template-columns: repeat(3, minmax(0, 1fr));
+}
+
+.catalog {
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: This is not pertinent to the applications list --> 🗑️

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
+  } @else {
+    <mat-card>
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]:
```suggestion
    <mat-card appearance="outlined">
```

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss [hunk 4]
```diff
@@ -1,15 +1,56 @@
...
+      }
+    }
+  }
+}
+
+mat-card {
+  padding: 16px;
+  border: 0.5px solid grey;
+  background-color: white;
+}
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: This should be 🗑️ . The `mat-card` should use the default values when assigned `appearance="outlined"` in the html tag.

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss [hunk 5]
```diff
@@ -1,15 +1,56 @@
...
+
+.padding-bottom {
+  padding-bottom: 16px;
+}
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: A little odd like this. I would instead make the parent container flex and add a gap of 16px.

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts [hunk 6]
```diff
@@ -13,13 +13,93 @@
...
+  bannerTitle: string;
+  bannerSubtitle: string;
+
+  private applicationService = inject(ApplicationService);
+  private page$ = new BehaviorSubject(1);
+
+  constructor(private configService: ConfigService) {
+    this.showBanner = this.configService.configuration?.portalNext?.banner?.enabled ?? false;
+    this.bannerTitle = this.configService.configuration?.portalNext?.banner?.title ?? '';
+    this.bannerSubtitle = this.configService.configuration?.portalNext?.banner?.subtitle ?? '';
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: 🗑️

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html [hunk 7]
```diff
@@ -1,18 +1,40 @@
...
+          <app-application-card [application]="application" />
+        }
+      </div>
+    </mat-card>
+  }
+}
+
+@if (loadingPage$ | async) {
+  <app-loader />
+}
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]:
```suggestion
<mat-card appearance="outlined">
   <mat-card-content>
      <h2>Applications</h2>
      <div class="padding-bottom">
        An application represents a developer's project that interacts with the API. It acts as a means to manage access control to APIs via
        subscriptions.
      </div>
   @if (applicationPaginator$ | async; as applications) {
      @if (applications.data.length === 0) {
        
       } @else {
        <div class="app-list__container" infiniteScroll (scrolled)="loadMoreApplications(applications)">
          @for (application of applications.data; track application.id) {
            <app-application-card [application]="application" />
          }
        </div>
      }
    }
   
    @if (loadingPage$ | async) {
      <app-loader />
    }
  </mat-card-content>
</mat-card>

```

This puts everything in `mat-card-content`, so the padding is handled for you. This also adds `appearance="outlined"` to the root `mat-card`.

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts [hunk 8]
```diff
@@ -13,13 +13,93 @@
...
   styleUrl: './applications.component.scss',
 })
-export class ApplicationsComponent {}
+export class ApplicationsComponent {
+  applicationPaginator$: Observable<ApplicationPaginatorVM>;
+  loadingPage$ = new BehaviorSubject(true);
+
+  showBanner: boolean;
+  bannerTitle: string;
+  bannerSubtitle: string;
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: 🗑️

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    <div class="app-card__description m3-body-medium">
+      @if (content) {
+        {{ content }}
+      } @else {
+        <span i18n="@@descriptionMissing">{{ missingContentMessage }}</span>
+      }
+    </div>
+  </mat-card-content>
+  <mat-card-actions class="app-card__actions">
+    <button i18n="@@cardButton" mat-stroked-button color="primary" [routerLink]="routerLinkValue">
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html]: The button text is dynamic, so the `@@cardButton` won't work. You can either create a directive or add it via `<ng-content>`

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        } @else {
+          <div class="m3-body-medium"><span i18n="@@owner">Owner: </span>{{ owner }}</div>
+        }
+      </div>
+    </div>
+    <div class="app-card__description m3-body-medium">
+      @if (content) {
+        {{ content }}
+      } @else {
+        <span i18n="@@descriptionMissing">{{ missingContentMessage }}</span>
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html]: `@@descriptionMissing` won't work since it's dynamic content.

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.ts [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+import { PictureComponent } from '../picture/picture.component';
+
+@Component({
+  selector: 'app-card',
+  standalone: true,
+  imports: [MatButton, MatCard, MatCardActions, MatCardContent, PictureComponent, RouterLink],
+  templateUrl: './app-card.component.html',
+  styleUrl: './app-card.component.scss',
+})
+export class AppCardComponent {
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.ts]: Not a huge fan of the name since it's confusing to have `ApplicationCardComponent` that exists as well. What about `TileCardComponent` or `CardListComponent` or ??

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html:1]: Given the fact that we have to do translations, I'm wondering if this shared component is more of a headache than a help. It's somewhat simple and would not be much code duplication between `ApplicationCardComponent` and `ApiCardComponent`. WDYT?

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.scss [hunk 12]
```diff
@@ -19,9 +19,10 @@
   flex-flow: column;
 }
 
-.api-card {
+.app-card {
   display: flex;
   height: 100%;
+  min-height: 256px;
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.scss]: This is ok, but it's best to avoid fixed heights in pixels for the CSS. If we ever want to create a mobile site (😱 ) or have a tablet-friendly site, it could make things difficult. If you'd like, we can 📞 to try to find a solution together. Otherwise, this can stay.

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html (hunk 1 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html (hunk 3 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss (hunk 4 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts (hunk 6 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html (hunk 7 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss (hunk 2 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts (hunk 8 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.ts (hunk 11 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss (hunk 5 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.scss]: Done

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html (hunk 9 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html]: Done

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html (hunk 10 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html]: Done

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.scss (hunk 12 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.scss]: Done

File: gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/app-card/app-card.component.html:1]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html [hunk 13]
```diff
@@ -1,18 +1,43 @@
...
 -->
-<p>applications works!</p>
+<mat-card appearance="outlined">
+  <mat-card-content class="tile">
+    <div>
+      <h2>Applications</h2>
+      <div class="description">
+        An application represents a developer's project that interacts with the API. It acts as a means to manage access control to APIs via
+        subscriptions.
+      </div>
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: add i18n

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts [hunk 14]
```diff
@@ -13,13 +13,91 @@
...
   styleUrl: './applications.component.scss',
 })
-export class ApplicationsComponent {}
+export class ApplicationsComponent {
+  applicationPaginator$: Observable<ApplicationPaginatorVM>;
+  loadingPage$ = new BehaviorSubject(true);
+
+  showBanner: boolean;
+  bannerTitle: string;
+
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: 🗑️

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts [hunk 15]
```diff
@@ -13,13 +13,91 @@
...
+  private applicationService = inject(ApplicationService);
+  private page$ = new BehaviorSubject(1);
+
+  constructor(private configService: ConfigService) {
+    this.showBanner = this.configService.configuration?.portalNext?.banner?.enabled ?? false;
+    this.bannerTitle = this.configService.configuration?.portalNext?.banner?.title ?? '';
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: 🗑️

File: gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.html [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+    <div class="app-card__description m3-body-medium">
+      @if (application.description) {
+        {{ application.description }}
+      } @else {
+        <span i18n="@@descriptionMissing">Description for this application is missing.</span>
+      }
+    </div>
+  </mat-card-content>
+  <mat-card-actions class="app-card__actions">
+    <button i18n="@@cardButton" mat-stroked-button color="primary" [routerLink]="['.', application.id]">Open</button>
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.html]: i18n id can be more specific to this component

File: gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.spec.ts [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+    expect(updatedCardHarnesses.length).toBe(1);
+  });
+
+  it('should display card component correctly with search query', async () => {
+    fixture.componentInstance.application.name = 'sample query';
+    fixture.detectChanges();
+
+    const cardHarnesses = await harnessLoader.getAllHarnesses(MatCardHarness);
+    expect(cardHarnesses.length).toBe(1);
+  });
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.spec.ts]: 🗑️  -- not pertinent to application-card component

**Review by jourdiw — CHANGES_REQUESTED** [review]

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.html (hunk 13 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.html]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts (hunk 14 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts (hunk 15 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.html (hunk 16 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.html]: Done

File: gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts (hunk 15 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/app/applications/applications.component.ts]: Done

File: gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.spec.ts (hunk 17 above)
Comment (illichso) [reply gravitee-apim-portal-webui-next/src/components/application-card/application-card.component.spec.ts]: Done

**Review by jourdiw — APPROVED** [review]
